ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.hpp

```diff
@@ -0,0 +1,28 @@
+
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+#include <functional>
+#include <optional>
+#include "tensor/tensor.hpp"
+#include "third_party/magic_enum/magic_enum.hpp"
+
+namespace ttnn::operations::binary{
+
+enum class BinaryCompositeOpType {
+    HYPOT,
+    XLOGY,
+    ADDALPHA,
+    SUBALPHA,
+    NEXTAFTER,
+    ISCLOSE,
+
+};
+struct BinaryCompositeFunction
+{
+    static std::function<Tensor(const Tensor&, const Tensor&, const std::optional<MemoryConfig>&)> get_function_type1(BinaryCompositeOpType OpType);
```

Comment:
@KalaivaniMCW @eyonland , do we wait till proper solution is established or will reiterate separately?

Reply:
@ayerofieiev-tt we'll start using the structs instead of std::function here onwards. avoids rework and saves lot of time. you're okay with #10169 changes too right?